hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3946,3 +3950,30 @@ func (r *HostedClusterReconciler) reconcileAWSSubnets(ctx context.Context, creat
 	}
 	return nil
 }
+
+// defaultAPIPortIfNeeded defaults the apiserver port on Azure management clusters as a workaround
+// for https://bugzilla.redhat.com/show_bug.cgi?id=2060650: Azure LBs with port 6443 don't work
+func (r *HostedClusterReconciler) defaultAPIPortIfNeeded(ctx context.Context, hcluster *hyperv1.HostedCluster) error {
+	infra := &configv1.Infrastructure{ObjectMeta: metav1.ObjectMeta{Name: "cluster"}}
```

Comment:
This would break non ocp management clusters. Why don't we infer the platform from the HC and also save the api server call?

Reply:
Good point with the non-ocp management clusters.

We can not infer it from the HC because the HC is not representative of this. Someone could have a an aws management cluster to use guest clusters or vice versa.

Reply:
Hmm, so the need finally arises for HO to be management cluster platform aware :thinking: We already have this if the HO is configured for private clusters in the `--private-platform` flag.  Maybe we need to change that to just a `--platform` flag if we need it regardless of private support.

Reply:
I am a big fan of discovering things automatically over requiring explicit config. That being said, we can use the infra resource to default the `--private-platform`, yes

Reply:
Right now I would mostly love to have something that fixes the issue at hand even if not perfect, because this blocks azure conforamnce testing which might discover yet unknown unknowns: https://github.com/openshift/release/pull/26666

Reply:
I didn't look at the full diff before to see the `ManagementClusterCapabilities` check beforehand.  This will not break on non-OCP management clusters.  It will `return nil` right from the start.

Reply:
The requirement comes from the management cluster, not the guest cluster. If the guest cluster imposes this requirement, then yes. I am not fully clear if something like AKS has the same requirement.